conformance/utils/suite/features.go

```diff
@@ -0,0 +1,163 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import "k8s.io/apimachinery/pkg/util/sets"
+
+// -----------------------------------------------------------------------------
+// Features - Types
+// -----------------------------------------------------------------------------
+
+// SupportedFeature allows opting in to additional conformance tests at an
+// individual feature granularity.
+type SupportedFeature string
+
+// -----------------------------------------------------------------------------
+// Features - Standard (Core)
+// -----------------------------------------------------------------------------
+
+const (
+	// This option indicates support for ReferenceGrant (core conformance).
+	// Opting out of this requires an implementation to have clearly implemented
+	// and documented equivalent safeguards.
+	SupportReferenceGrant SupportedFeature = "ReferenceGrant"
+)
+
+// StandardCoreFeatures are the features that are required to be conformant with
+// the Core API features (e.g. GatewayClass, Gateway, e.t.c.).
+//
+// TODO: rename this from "Standard" as that might be confusing with our
+// "Standard" release channel.
+var StandardCoreFeatures = sets.New(
+	SupportReferenceGrant,
+)
+
+// -----------------------------------------------------------------------------
+// Features - Standard (Extended)
+// -----------------------------------------------------------------------------
+
+const (
+	// This option indicates GatewayClass will update the observedGeneration in
+	// it's conditions when reconciling (extended conformance).
+	//
+	// TODO: we intend to make this core and require implementations to do it
+	//       as we expect this is something every implementation should be able
+	//       to do and it's ideal behavior.
+	//
+	//       See: https://github.com/kubernetes-sigs/gateway-api/issues/1780
+	SupportGatewayClassObservedGenerationBump SupportedFeature = "GatewayClassObservedGenerationBump"
+
+	// This option indicates support for Destination Port matching.
+	SupportRouteDestinationPortMatching SupportedFeature = "RouteDestinationPortMatching"
+)
+
+// StandardExtendedFeatures are extra generic features that implementations may
+// choose to support as an opt-in.
+//
+// TODO: rename this from "Standard" as that might be confusing with our
+// "Standard" release channel.
+var StandardExtendedFeatures = sets.New(
+	SupportGatewayClassObservedGenerationBump,
+	SupportRouteDestinationPortMatching,
+).Insert(StandardCoreFeatures.UnsortedList()...)
```

Comment:
Similar to my comment above, I'd rather punt on this standard categorization until we have a clearer picture of the common buckets. I think both of these features could be closely tied to individual resources instead (GatewayClass and Gateway respectively). It could help with our test output if we could show that an implementation supports 80% of extended features on Gateway for example.

Reply:
So in case it wasn't clear this isn't exactly a "new" thing, something similar exists currently:

```go
// StandardCoreFeatures are the features that are required to be conformant with
// the Core API features that are part of the Standard release channel.
var StandardCoreFeatures = sets.New(
	SupportReferenceGrant,
)
```

https://github.com/kubernetes-sigs/gateway-api/blob/004eb1b5f290431dd746ee8b1ed21d63d80ac5c7/conformance/utils/suite/suite.go#L77

So I'm not clear on how we would "punt" on this, which seems to insinuate we don't add it at all without completely removing something that existed before in any fashion. Is that what you're advocating for? You just want us to completely remove it for now?

Reply:
Yeah you're right, I missed that `StandardCoreFeatures` already exists. It feels weird to have `StandardCoreFeatures` be so empty, but I guess this is really serving as a list of things we're allowing implementations to skip when running conformance tests, even though they're core. I guess this all makes sense, it just feels especially weird with the overlap with standard channel. I hate to merge code that has open TODOs without issues to track them, so if we want to stick with the `standard` name here I think we at least need a follow up issue to come up with a better one.

Reply:
Sounds good, https://github.com/kubernetes-sigs/gateway-api/issues/1891 was created and the TODO items have been updated to link it.